db/view/view_builder.hh

```diff
@@ -198,7 +199,7 @@ public:
      * Requires that all views have been loaded from the system tables and are accessible
      * through the database, and that the commitlog has been replayed.
      */
-    future<> start(service::migration_manager&);
+    future<> start(service::migration_manager&, utils::cross_shard_barrier b = utils::cross_shard_barrier(utils::cross_shard_barrier::solo{}));
```

Comment:
Why pass a barrier which turns synchronization off by default?

For example, this function is called in `cql_test_env.cc`:

```c++
            _view_builder.invoke_on_all([this] (db::view::view_builder& vb) {
                return vb.start(_mm.local());
            }).get();
```

I'm not sure turning synchronization off there makes sense. I think that a solo barrier is a dangerous choice for a default parameter.

Reply:
I made it solo by default, because all users but main run in single thread, even cql test env. The same approach is used with replica::database, it uses solo barrier by default. However, database is instantiated in more places than view builder, so using default and not patching those places saves some code. With view builder it's only main and test, so I can patch cql test env to be explicit, if you insist.

Reply:
> because all users but main run in single thread, even cql test env.

Are you sure about this? I can see at least one boost test which uses `do_with_cql_env_thread` but skips itself unless it was run with more than one shard (`test_system_config_table_update` in `virtual_table_test.cc`).